source/code/source/playfab/PlayFabAuthenticationContext.cpp

```diff
@@ -7,27 +7,42 @@ namespace PlayFab
 {
     PlayFabAuthenticationContext::PlayFabAuthenticationContext()
     {
-#ifndef DISABLE_PLAYFABCLIENT_API
-        clientSessionTicket = PlayFabSettings::clientSessionTicket;
-#endif
-#ifndef DISABLE_PLAYFABENTITY_API
-        entityToken = PlayFabSettings::entityToken;
-#endif
-#if defined(ENABLE_PLAYFABSERVER_API) || defined(ENABLE_PLAYFABADMIN_API)
-        developerSecretKey = PlayFabSettings::developerSecretKey;
-#endif
+        ForgetAllCredentials();
     }
 
     void PlayFabAuthenticationContext::ForgetAllCredentials()
     {
 #ifndef DISABLE_PLAYFABCLIENT_API
+        playFabId.clear();
         clientSessionTicket.clear();
+        advertisingIdType.clear();
+        advertisingIdValue.clear();
 #endif
-#ifndef DISABLE_PLAYFABENTITY_API
+        entityId.clear();
+        entityType.clear();
         entityToken.clear();
-#endif
-#if defined(ENABLE_PLAYFABSERVER_API) || defined(ENABLE_PLAYFABADMIN_API)
-        developerSecretKey.clear();
-#endif
     }
-}
\ No newline at end of file
+
+    void SetIfNotNull(const std::string& input, std::string& output)
+    {
+        if (!input.empty())
```

Comment:
>if (!input.empty()) [](start = 8, length = 19)

what if input is null ? should we not check for it ?

Reply:
input is a std::string reference so there is no concern of null. A default constructed std::string will just be empty not null

Reply:
Passed by ref, it shouldn't be null.
By the signature and usage here in this file, it should never be null.
It's technically possible, but you'd have to violate the signature of the function to accomplish it.

---
In reply to: [343405485](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r343405485) [](ancestors = 343405485)